torch/nn/quantized/dynamic/modules/linear.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import torch
+from ....modules.module import Module
+from ....modules.linear import Linear as NNLinear
+
+class Linear(NNLinear):
```

Comment:
you can inherit from nnq.Linear if you want to reuse `.from_float`

Reply:
Then we need to add some logic to determine whether we should add the activation observer?

Reply:
Yes, please inherit from nnq.Linear. This way, you inherit loading/saving of state dict.  I would recommend the following changes:

```python
class Linear(nn.quantized.Linear):
  
    __constants__ = ['bias', 'in_features', 'out_features']
    def __init__(self, in_features, out_features, bias=True):
        super(Linear, self).__init__(in_features, out_features, bias)
           if bias:
             del self.bias
             bias= torch.Tensor(out_features)
             self.register_buffer('bias', bias)
        else:
            self.register_buffer('bias', None)
      del self.scale
      del self.zero_point

    def forward(self, x):
        # Note that we can handle self.bias == None case.
        Y = torch.ops.quantized.fbgemm_linear_dynamic(
            x, self._packed_weight,
            self.bias)
        return Y.to(x.dtype)

 @staticmethod
    def from_float(mod):
        r"""Create a quantized module from a float module or qparams_dict
            Args: `mod` a float module, either produced by torch.quantization utilities
            or directly from user
        """
   
            assert type(mod) == NNLinear, 'nnq.Linear.from_float only works for nn.Linear'
            assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
            weight_observer = mod.qconfig.weight()
            weight_observer(mod.weight)
           wt_scale, wt_zp = weight_observer.calculate_qparams()
           qweight = torch.quantize_linear(mod.weight.float(), wt_scale, wt_zp.long().item(), torch.qint8)
           qlinear = Linear(mod.in_features, mod.out_features, mod.bias) # Instantiate dynamic.linear module
           qlinear._packed_weight = torch.ops.quantized.fbgemm_linear_prepack(qweight)
         # Can handle bias being None by just copying it to qLinear.bias
          qlinear.bias = mod.bias
     
         return qlinear

Reply:
I'm not sure inheriting and killing so many attributes is worth it. Maybe Static QLinear should inherit from this one instead? (or have common base class).

I'm also fine with duplication of getstate/setstate. We likely can simplify it further once custom C++ classes are fully ready.

Reply:
I currently adopted the approach of inheriting from `nnq.Linear`, as this one is more straightforward. Of course a better way is to have the common base class.

Reply:
Please land this version first and create a follow up task for refactoring Linear to have a common base class.

Reply:
Filed a task.